cmd/podman/create.go

```diff
@@ -67,15 +67,33 @@ func createCmd(c *cli.Context) error {
 		}
 	}
 
-	if len(c.Args()) < 1 {
+	rootfs := c.String("rootfs")
+	if len(c.Args()) < 1 && rootfs == "" {
 		return errors.Errorf("image name or ID is required")
 	}
 
 	mappings, err := util.ParseIDMapping(c.StringSlice("uidmap"), c.StringSlice("gidmap"), c.String("subuidmap"), c.String("subgidmap"))
 	if err != nil {
 		return err
 	}
+	if mappings.UIDMap == nil && os.Getuid() != 0 {
+		uids := []string{fmt.Sprintf("0:%d:1", os.Getuid())}
+		gids := []string{fmt.Sprintf("0:%d:1", os.Getgid())}
+		c.Set("uidmap", uids[0])
+		c.Set("gidmap", gids[0])
+		mappings, err = util.ParseIDMapping(uids, gids, "", "")
+		if err != nil {
+			return err
+		}
+	}
 	storageOpts := storage.DefaultStoreOptions
+	if os.Getuid() != 0 {
+		var err error
+		storageOpts, err = libpodruntime.GetRootlessStorageOpts()
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
(3 copies of identical `DefaultStoreOptions`+`GetRootlessStorageOpts` — 3 copies is my rough threshold for almost certainly benefiting from a helper function.)

Reply:
I've added a helper function